src/utils/time.ts

```diff
@@ -116,9 +116,9 @@ function getSunsetSunriseUTCTime(
     latitude: number,
     longitude: number,
 ) {
-    const dec31 = new Date(date.getUTCFullYear(), 0, 0);
```

Comment:
This is the buggy line. The problem is `new Date(...)` returns an object which internally keeps track of user's time zone. Instead, we can use `Date.UTC(...)` which just returns a timestamp in UTC GMT which does not depend on user's time zone.